### PR TITLE
fixes in satellite-change-hostname tests

### PR DESCRIPTION
**Pull request by pondrejk**
This PR adds an /etc/hosts entry to pass s-c-h post-rename validation, negative tests now check for hostname initially

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_rename.py

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_rename.py

**Comment by pondrejk** [comment]
from rpt:

```
13:39:12  ============================= test session starts ==============================
13:39:12  platform linux -- Python 3.8.12, pytest-7.1.2, pluggy-1.0.0
13:39:12  shared_function enabled - OFF - scope:  - storage: file
13:39:12  rootdir: /opt/app-root/src/robottelo, configfile: pyproject.toml
13:39:12  plugins: forked-1.4.0, ibutsu-2.2.2, mock-3.8.2, reportportal-5.1.2, services-2.2.1, xdist-2.5.0
13:39:12  collected 5 items / 1 deselected / 4 selected
13:39:12  
14:45:11  tests/foreman/destructive/test_rename.py F...                            [100%]
```

The one failure occurred in the positive test well after successful rename, seems that grep exited with an unknown error

File: tests/foreman/destructive/test_rename.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def test_negative_rename_sat_wrong_passwd(module_target_sat):
     :CaseAutomation: Automated
     """
     username = settings.server.admin_username
-    original_name = module_target_sat.hostname
+    original_name = module_target_sat.execute('hostname').stdout.strip()
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_rename.py:191]: ` module_target_sat.hostname` doesn't work here? any reason to run hostname cmd explicitly?

File: tests/foreman/destructive/test_rename.py [hunk 2]
```diff
@@ -71,6 +71,11 @@ def test_positive_rename_satellite(module_org, module_product, module_target_sat
     medium_path = f'http://{old_hostname}/testpath-{gen_string("alpha")}/os/'
     medium = module_target_sat.api.Media(organization=[module_org], path_=medium_path).create()
     repo = module_target_sat.api.Repository(product=module_product, name='testrepo').create()
+    # create /etc/hosts entry to pass s-c-h validation
+    sat_ip = module_target_sat.execute(
+        "ip addr show eth0 | grep 'inet ' | awk '{print $2}' | cut -d/ -f1"
+    ).stdout.strip()
```
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/destructive/test_rename.py:77]: tried that out but didn't work, not sure if that's supposed to work for a host that was not explicitly registered (the underlying satellite)

File: tests/foreman/destructive/test_rename.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/destructive/test_rename.py:191]: yeah, as these test operate on the same "destructive" sat instance, the negative tests can get to the satellite already renamed by the positive tests, hence this change

**Review by latran — COMMENTED** [review]
Have you tried fixing that one failure from the test run?  Seems to be an assertion error.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_rename.py -k test_positive_rename_satellite

**Review by mshriver — APPROVED** [review]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_rename.py -k test_positive_rename_satellite

**Comment by pondrejk** [comment]
@latran the assertion error was due to grep not being able to access some files, so it exited with code 2 instead of 1, adjusted the assertion

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_rename.py -k test_positive_rename_satellite

**Pull request merged**